flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/FlinkCalciteCatalogReader.java

```diff
@@ -104,6 +111,30 @@ public Prepare.PreparingTable getTable(List<String> names) {
         }
     }
 
+    public CatalogSchemaModel getModel(List<String> names) {
+        for (List<String> schemaPath : getSchemaPaths()) {
+            CalciteSchema schema =
+                    getSchema(
+                            getRootSchema(),
+                            Iterables.concat(schemaPath, Util.skipLast(names)),
+                            nameMatcher());
+            if (schema == null) {
+                continue;
+            }
+
+            if (!(schema.schema instanceof FlinkSchema)) {
+                throw new ValidationException("getModel() only supports FlinkSchema.");
+            }
+
+            FlinkSchema flinkSchema = (FlinkSchema) schema.schema;
+            CatalogSchemaModel model = flinkSchema.getModel(Util.last(names));
```

Comment:
Flink pasrer supports to parse statement in case insensitive/case sensitive mode. It's better we can align the behaviour as getTable. cc SimpleCalciteSchema#getImplicitTable

Reply:
Good call